selfservice/hook/web_hook.go

```diff
@@ -604,31 +624,62 @@ func (e *WebHook) parseWebhookResponse(resp *http.Response, id *identity.Identit
 		return nil
 	} else if resp.StatusCode >= http.StatusBadRequest {
 		var hookResponse rawHookResponse
-		// todo: replace it with json.NewDecoder(resp.Body).Decode(&hookResponse) to be consistent with upstream
-		// used json.Unmarshal for now, because json.NewDecoder(resp.Body).Decode(&hookResponse) panics
+		var hookResponseFallback rawHookResponseFallback
 		if err = json.Unmarshal(body, &hookResponse); err != nil {
 			return errors.Wrap(err, "webhook response could not be unmarshalled properly from JSON")
 		}
 
+		// try to read another format is we could not read detailed messages
+		if len(hookResponse.Messages) > 0 && len(hookResponse.Messages[0].DetailedMessages) < 1 {
+			if err = json.Unmarshal(body, &hookResponseFallback); err != nil {
+				return errors.Wrap(err, "webhook response could not be unmarshalled properly from JSON")
+			}
+		}
+
 		var validationErrs []*schema.ValidationError
-		for _, msg := range hookResponse.Messages {
-			messages := text.Messages{}
-			for _, detail := range msg.DetailedMessages {
-				var msgType text.UITextType
-				if detail.Type == "error" {
-					msgType = text.Error
-				} else {
-					msgType = text.Info
+
+		// fandom-start - letter case workaround
+		if len(hookResponse.Messages) > 0 && len(hookResponse.Messages[0].DetailedMessages) > 0 {
```

Comment:
Nitpick: comparing to 0 is more readable
```suggestion
		if len(hookResponse.Messages) != 0 && len(hookResponse.Messages[0].DetailedMessages) != 0 {
```